src/rust/base/src/config.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019 Shift Cryptosecurity AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::time::Duration;
+use crate::error::Error;
+use crate::screen::print_debug;
+use arrayvec::ArrayString;
+
+pub struct Config {
+    pub(crate) hostname: Option<ArrayString<[u8;64]>>,
+}
+
+impl Config {
+    pub const fn new() -> Config {
+        Config {
+            hostname: None,
+        }
+    }
+
+    pub fn set_hostname(&mut self, hostname: &str) -> Result<(), Error>  {
+        if hostname.chars().any(|c| !c.is_alphanumeric()) {
+            return Err(Error::InvalidHostname);
+        }
+        // We need to use unsafe to access globals mutably
```

Comment:
what is unsafe? where is the mutable global? sorry, i'm new to this :P

Reply:
stray comment. Moved the code, forgot about the comment ;)